zookeeper-server/src/main/java/org/apache/zookeeper/server/admin/JettyAdminServer.java

```diff
@@ -89,17 +90,20 @@ public JettyAdminServer() throws AdminServerException, IOException, GeneralSecur
             System.getProperty("zookeeper.admin.commandURL", DEFAULT_COMMAND_URL),
             Integer.getInteger("zookeeper.admin.httpVersion", DEFAULT_HTTP_VERSION),
             Boolean.getBoolean("zookeeper.admin.portUnification"),
-            Boolean.getBoolean("zookeeper.admin.forceHttps"));
+            Boolean.getBoolean("zookeeper.admin.forceHttps"),
+            Boolean.getBoolean("zookeeper.admin.needClientAuth"));
     }
 
+    @SuppressWarnings("deprecation")
```

Comment:
Here is a patch to reduce complex:

```diff
diff --git a/zookeeper-server/src/main/java/org/apache/zookeeper/server/admin/JettyAdminServer.java b/zookeeper-server/src/main/java/org/apache/zookeeper/server/admin/JettyAdminServer.java
index 2c3bb723d..a237e4c3b 100644
--- a/zookeeper-server/src/main/java/org/apache/zookeeper/server/admin/JettyAdminServer.java
+++ b/zookeeper-server/src/main/java/org/apache/zookeeper/server/admin/JettyAdminServer.java
@@ -94,7 +94,6 @@ public JettyAdminServer() throws AdminServerException, IOException, GeneralSecur
Boolean.getBoolean("zookeeper.admin.needClientAuth"));
}

-    @SuppressWarnings("deprecation")
public JettyAdminServer(
String address,
int port,
@@ -148,15 +147,12 @@ public JettyAdminServer(
throw e;
}

-                SslContextFactory sslContextFactory = new SslContextFactory.Server();
+                SslContextFactory.Server sslContextFactory = new SslContextFactory.Server();
sslContextFactory.setKeyStore(keyStore);
sslContextFactory.setKeyStorePassword(privateKeyPassword);
sslContextFactory.setTrustStore(trustStore);
sslContextFactory.setTrustStorePassword(certAuthPassword);
-                if (needClientAuth) {
-                    // setNeedClientAuth() of SslContextFactory.Server is not a deprecated API
-                    sslContextFactory.setNeedClientAuth(true);
-                }
+                sslContextFactory.setNeedClientAuth(needClientAuth);

if (forceHttps) {
connector = new ServerConnector(server,
```

Reply:
Thanks for this. It's much simpler and clean now.

Updated.